Makefile

```diff
@@ -144,21 +149,16 @@ controller-gen: $(CONTROLLER_GEN) ## Download controller-gen locally if necessar
 $(CONTROLLER_GEN): $(LOCALBIN)
 	test -s $(LOCALBIN)/controller-gen || GOBIN=$(LOCALBIN) go install sigs.k8s.io/controller-tools/cmd/controller-gen@$(CONTROLLER_TOOLS_VERSION)
 
+.PHONY: lint
+lint: golangci-lint
+	 $(GOLANGCI_LINT) run --issues-exit-code $(GOLANGCI_EXIT_CODE)
+
 # Run go fmt against code
 .PHONY: fmt
-fmt:
-	go fmt ./...
-
-# Run go vet against code
-.PHONY: vet
-vet:
-	go vet ./...
-
-# Run https://staticcheck.io against code
-.PHONY: staticcheck
-staticcheck:
-	go install honnef.co/go/tools/cmd/staticcheck@latest
-	staticcheck ./...
+fmt: golangci-lint ## Ensure consistent code style
+	@go mod tidy
+	@go fmt ./...
+	@$(GOLANGCI_LINT) run --fix --issues-exit-code $(GOLANGCI_EXIT_CODE)
```

Comment:
I believe `fmt: golangci-lint` will run the `lint` task and then this `fmt` task, which itself also runs the linter. I don't think we should run the linter as a part of the `fmt` task since its job is to format and not lint (i.e. separate concerns). 

I suggest removing the linter from this `fmt` task and run the linter as a part of `make test` or `make test-with-vendor` as was previously done with `vet staticcheck`. We can run the linter separately as you have configured with `make lint`